### ISSUE TITLE
Global beforeHooks

### ISSUE THREAD

**Issue by danielo515** [jeffbski/bench-rest#25]
Hello,

Sometimes there are some hooks that I would like to run for every request on the stream (for example, for include certain common header).

Currently there is no way to do this. At first instance I though that `beforeMain` was for that, but this is a mistake from my side ,probably because the total absence of a clear usage example of `beforeMain`. BeforeMain should be a list of REST request, probably for getting some cookie or something like that.

Would it be possible to impelemnt a beforeHooks configurable at flow level insted of just request?

Thanks and regards

**Comment by jeffbski** [comment]
Sorry for the delay in getting back to you. Yes it would be possible, maybe we'd add a beforeEach and afterEach hooks that would run immediately before and after each request.

Would that work for your use case?

**Comment by danielo515** [comment]
Yes, that will definitely fit my use-case. But some data may be required, like the request, a callback and that stuff.
